### PR TITLE
added login and logout for user

### PR DESCRIPTION
**Pull request by ImennoArtemiy**

### CODE REVIEW

File: src/ducks/user/reducers.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+export const userReducer = (state = initialState, action) => {
+    if (action.type === LOGIN) {
+        return {...state, isLogged: true}
+    }
+    if (action.type === LOGOUT) {
+        return {...state, isLogged: false}
+    }
+
+    return {...state}
```
Comment (romos21) [inline src/ducks/user/reducers.js]: it's better to use switch case . But also redux tollkit can halp you

File: src/views/AuthPage/AuthPage.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import Header from "../TopSection/Header/Header";
+import FormAuth from "./FormAuth/FormAuth";
+import {useSelector} from "react-redux";
+import {Navigate} from "react-router-dom";
+
+function AuthPage () {
+
+    const isLogin = useSelector(state => state.user.isLogged)
+
+    return isLogin === false ? (<section className='topSection'>
```
Comment (romos21) [inline src/views/AuthPage/AuthPage.js]: you can replace isLogin === false ? with isLogin ?

File: src/views/MainPage/MainPage.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    const [serverData, setServerData] = useState([])
+    const [value, setValue] = useState('0')
+
+    useEffect(() => {
+
+        fetch('https://fe-student-api.herokuapp.com/api/hotels/popular')
+            .then((response) => {
+                return response.json()
+            })
+            .then(setServerData)
```
Comment (romos21) [inline src/views/MainPage/MainPage.js:19]: does it work? Im not sure case you don't call this function

File: src/views/MainPage/MainPage.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    }, [])
+
+
+    useEffect(() => {
+
+        fetch(`https://fe-student-api.herokuapp.com/api/hotels?search=${value}`)
+            .then((response) => {
+                return response.json()
+            })
+            .then(setSearchValue)
```
Comment (romos21) [inline src/views/MainPage/MainPage.js:29]: does it work? Im not sure case you don't call this function

File: src/views/TopSection/Header/Header.js [hunk 5]
```diff
@@ -1,14 +1,24 @@
...
 
-function Header ({wrapperStyle}) {
+function Header ({wrapperStyle, accountStyle}) {
+
+    const isLogin = useSelector(state => state.user.isLogged)
+    const [wantsToOut, setWantsToOut] = useState(false)
+
+    const accountClick = () => {
+        wantsToOut ? setWantsToOut(false) : setWantsToOut(true)
+    }
```
Comment (romos21) [inline src/views/TopSection/Header/Header.js]: setWantsToOut(!wantsToOut)

File: src/views/MainPage/MainPage.js (hunk 3 above)
Comment (ImennoArtemiy) [reply src/views/MainPage/MainPage.js:19]: it works)

File: src/views/MainPage/MainPage.js (hunk 4 above)
Comment (ImennoArtemiy) [reply src/views/MainPage/MainPage.js:29]: it works)

**Pull request merged**